processing/src/main/java/io/druid/segment/QueryableIndexIndexableAdapter.java

```diff
@@ -180,10 +181,12 @@ public int indexOf(String value)
           boolean done = false;
 
           {
-            dimensions = Maps.newLinkedHashMap();
+            final ArrayList<DictionaryEncodedColumn> dictionaryEncodedColumnArrayList = new ArrayList<>();
```

Comment:
You can use Lists.newArrayListWithCapacity to initialize the array with the right capacity